File: Api/src/permission/domain/entity/permission.go
package entity

type AddPermissionDTO struct {
	ID     int
	UserId int `gorm:"column:id_user" json:"id_user"`
	AppId  int `gorm:"column:id_app" json:"id_app"`
	RoleId int `gorm:"column:id_role" json:"id_role"`
}

type UpdatePermissionDTO struct {
	AppId  int `gorm:"column:id_app" json:"id_app"`
	RoleId int `gorm:"column:id_role" json:"id_role"`
}

type Permission struct {
	ID          int
	UserId      int    `gorm:"column:id_user" json:"id_user"`
	AppId       int    `gorm:"column:id_app" json:"id_app"`
	RoleId      int    `gorm:"column:id_role" json:"id_role"`
	Username    string `json:"username"`
	AppName     string `json:"app_name"`
	DisplayName string `json:"display_name"`
	Role        string `json:"role"`
}

File: frontend/lib/types/models/permission/permission.ts
export default class Permission {
    ID: number;
    id_user: number;
    id_app: number;
    id_role: number;

    constructor(ID = 0, id_user = 0, id_app = 0, id_role = 0) {
        this.ID = ID;
        this.id_user = id_user;
        this.id_app = id_app;
        this.id_role = id_role;
    }
}
